From 051748825e77172677d9e3b319b870c3c0a70a38 Mon Sep 17 00:00:00 2001
From: Jitka Plesnikova <jplesnik@redhat.com>
Date: Fri, 14 Jul 2017 14:13:50 +0200
Subject: [PATCH] Fix build failures for MariaDB 10.2
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

<mariadb_version.h> does not exist in mariadb-5.5.26. Do not include
it explicitly. Instead rely on including <mysql.h> that transitively
includes <mariadb_version.h> or <mysql_version.h>. This makes this
patch more portable.

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 dbdimp.c | 7 +++++++
 mysql.xs | 4 ++--
 2 files changed, 9 insertions(+), 2 deletions(-)

# Fixed in 4.045
#diff --git a/dbdimp.c b/dbdimp.c
#index 71251da..97fa9c4 100644
#--- a/dbdimp.c
#+++ b/dbdimp.c
#@@ -2104,6 +2104,9 @@ MYSQL *mysql_dr_connect(
# 
#     if (result)
#     {
#+#if MYSQL_VERSION_ID >= 50013
#+       my_bool reconnect= 1;
#+#endif
# #if MYSQL_VERSION_ID >=SERVER_PREPARE_VERSION
#       /* connection succeeded. */
#       /* imp_dbh == NULL when mysql_dr_connect() is called from mysql.xs
#@@ -2122,7 +2125,11 @@ MYSQL *mysql_dr_connect(
#         we turn off Mysql's auto reconnect and handle re-connecting ourselves
#         so that we can keep track of when this happens.
#       */
#+#if MYSQL_VERSION_ID >= 50013
#+      mysql_options(result, MYSQL_OPT_RECONNECT, &reconnect);
#+#else
#       result->reconnect=0;
#+#endif
#     }
#     else {
#       /* 
diff --git a/mysql.xs b/mysql.xs
index 60cf9c6..750c763 100644
--- a/mysql.xs
+++ b/mysql.xs
@@ -787,7 +787,7 @@ dbd_mysql_get_info(dbh, sql_info_type)
     D_imp_dbh(dbh);
     IV type = 0;
     SV* retsv=NULL;
-#if !defined(MARIADB_BASE_VERSION) && MYSQL_VERSION_ID >= 50709
+#if MYSQL_VERSION_ID >= 50709
 /* MariaDB 10 is not MySQL source level compatible so this only applies to MySQL*/
     IV buffer_len;
 #endif 
@@ -819,7 +819,7 @@ dbd_mysql_get_info(dbh, sql_info_type)
 	    retsv = newSVpvn("`", 1);
 	    break;
 	case SQL_MAXIMUM_STATEMENT_LENGTH:
-#if !defined(MARIADB_BASE_VERSION) && MYSQL_VERSION_ID >= 50709
+#if MYSQL_VERSION_ID >= 50709
         /* MariaDB 10 is not MySQL source level compatible so this
            only applies to MySQL*/
 	    /* mysql_get_option() was added in mysql 5.7.3 */
-- 
2.13.6

